content/javascript/concepts/requests/requests.md

```diff
@@ -0,0 +1,118 @@
+---
+Title: "Requests"
+Subjects:
+  - "Web Development"
+  - "Computer Science"
+Tags: 
+  - "Requests"
+CatalogContent:
+  - "introduction-to-javascript"
+  - "paths/front-end-engineer-career-path"
+---
+
+`XMLHttpRequest` is a built-in browser object that allows to make HTTP requests in JavaScript.
+
+It is basically used to fetch data from API's.
+
+There’s another, modern method `fetch`, that somewhat deprecates `XMLHttpRequest` but the focus of this post will be on `XMLHttpRequest` only.
+
+## Getting Started
+
+`XMLHttpRequest` can be used in two modes: synchronous and asynchronous.
+
+### 1. Asynchronous Method
+
+There are 3 steps to do this:
+
+1. Create `XMLHttpRequest`:
+```js
+let xhr = new XMLHttpRequest();
+```
+The constructor has no arguments.
+
+2. Initialize it,:
+```js
+xhr.open(method, URL, [async, user, password])
+```
+
+This method specifies the main parameters of the request:
+
+- `method` – HTTP-method. Usually `"GET"` or `"POST"`.
+- `URL` – the URL to request, a string, can be URL object.
+- `async` – if explicitly set to `false`, then the request is synchronous.
+- `user`, `password` – login and password for basic HTTP auth (if required).
+
+Please note that `open` call, contrary to its name, does not open the connection. It only configures the request, but the network activity only starts with the call of `send`.
+
+3. Send it out.
+```js
+xhr.send([body])
+```
+
+This method opens up the connection and sends the request to server. The optional body parameter contains the request `body`.
+
+Some request methods like `GET` do not have a `body`. And some of them like `POST` use `body` to send the data to the server.
+
+4. Listen to xhr events for response.
+
+These three events are the most widely used:
+
+- `load`– when the request is complete and the response is fully downloaded.
```

Comment:
Let's do colon : instead of dash - for all of these bullet points.

- `load`: When the request is complete and the response is fully downloaded.